desktop/window-handlers/spellcheck/index.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+/**
+ * External dependencies
+ */
+const { Menu, MenuItem } = require( 'electron' );
+
+module.exports = function( mainWindow ) {
+	mainWindow.webContents.on( 'context-menu', ( event, params ) => {
+		const menu = new Menu();
+
+		const copy = new MenuItem( { label: 'Copy', role: 'copy' } );
+
+		if ( !params.isEditable ) {
+			// If text is not editable, only permit the `Copy` action
+			menu.append( copy );
+		} else {
+			// Add each spelling suggestion
+			for ( const suggestion of params.dictionarySuggestions ) {
+				menu.append( new MenuItem( {
+					label: suggestion,
+					click: () => mainWindow.webContents.replaceMisspelling( suggestion )
+				} ) );
+			}
+
+			// Allow users to add the misspelled word to the dictionary
+			if ( params.misspelledWord ) {
+				menu.append( new MenuItem( { type: 'separator' } ) );
+				menu.append(
+					new MenuItem( {
+						label: 'Add to Dictionary',
+						click: () => mainWindow.webContents.session.addWordToSpellCheckerDictionary( params.misspelledWord ),
```

Comment:
does this persist across app runs?

Reply:
@dmsnell Yep - it does. This will add suggestions to the user's system dictionary, so the changes will be available from that point on. Words added to the dictionary will be available from that point on (and in fact be available to other applications once added).

Reply:
This is nice! This does not work with electron spellchecker.